Buttons

Hadj-Said-Bouras opened the pull request.

Hadj-Said-Bouras committed:
feat(buttons.jsx): create buttons without any functions only ui

Buttons #34

Hadj-Said-Bouras committed:
feat(buttons.jsx): add uk flag as a button

Buttons #34

ismail-benlaredj commented:
Hey Hadj it looks like the pull request was targeted to merge into the main branch, please change the target branch for the pull request to **dev**

Hadj-Said-Bouras commented:
changed to dev branch

darinetag reviewed (approved):
Job well done.

ismail-benlaredj commented on src/components/Buttons/Buttons.jsx:
Nice job, Hadj 🚀 But you can make your code even shorter in React There's a cool thing  in React called "reusable components" it means you can use the same code for different variation.

Instead of writing lots of code for each button, you can make one button work for everything for example you can pass only one function as prop (onClick function) and when we want use button we will pass the function that we need also in the props, let's say we have **buttonType** props you might set "buttonType" to 'buttonWithIcon'. If you provide an icon, the component will create a button with that icon. If you set "buttonType" to 'simple' and don't give it an icon (leave it undefined), it'll return a plain button with only text
This is a fundamental idea in React Once you get it, React becomes a easy.

Hadj-Said-Bouras replied:
Thank you for your feedback!

Hadj-Said-Bouras committed:
feat(index.jsx): sende functions icons and names as props

ismail-benlaredj committed:
Merge branch 'dev' into buttons

ismail-benlaredj committed:
feat(button component): add clsx, tailwind-merge, class-variance-authority and cn function

Added libraries: clsx, tailwind-merge, class-variance-authority, and created a 'cn' function
(lib/utils) that allows using all three libraries together.

ismail-benlaredj committed:
feat(button component): add button with different variation

feat #32
